src/parenttext_pipeline/configs.py

```diff
@@ -0,0 +1,198 @@
+    "qr_treatment" : QRTreatmentStepConfig,
+    "load_flows": StepConfig,
+    "extract_texts_for_translators": StepConfig,
+    "fix_arg_qr_translation": StepConfig,
+    "has_any_word_check": StepConfig,
+    "overall_integrity_check": StepConfig,
+}
+
+
+@dataclass(kw_only=True)
+class SourceConfig:
+    # Format of the source data
+    format: str
+
+
+@dataclass(kw_only=True)
+class SheetsSourceConfig(SourceConfig):
+    # Input format of the sheets.
+    # Either google_sheets, csv, json or xlsx
+    subformat: str
+    # For each `(name, filepath)` entry in `{files_dict}`, the processed version
+    # of `{filepath}` is stored as `{name}.json`.
+    files_dict: dict[str, str] = field(default_factory=dict)
+
+    # If files_archive is None: List of Google Sheet IDs to read from
+    # If files_archive is not None: List of folder names within archive
+    files_list: list[str] = field(default_factory=list)
+    # Path or URL to a zip archive containing folders
+    # each with sheets in CSV format (no nesting)
+    files_archive: str = None
+
+
+@dataclass(kw_only=True)
+class JSONSourceConfig(SourceConfig):
+    # For each `(name, filepath)` entry in `{files_dict}`, the processed version
+    # of `{filepath}` is stored as `{name}.json`.
+    files_dict: dict[str, str]
+
+
+@dataclass(kw_only=True)
+class SafeguardingSourceConfig(SourceConfig):
+    # Either filepath or sources has to be provided
+
+    # Path to json file with safeguarding words
+    filepath: str = None
+    # List of XLSX files with safeguarding words
+    # Each source is a dict with two entries:
+    #     Key: 3-letter language key
+    #     path: path to an XLSX file containing safeguarding words
+    sources: list[dict[str, str]] = None
+
+    def __post_init__(self):
+        if self.filepath is None and self.sources is None:
+            raise ValueError("For SafeguardingSourceConfig, either filepath or sources needs to be provided")
+
+
+@dataclass(kw_only=True)
+class TranslationSourceConfig(SourceConfig):
+    # Languages for which to pull the translation data.
+    # Each entry is a dict with two keys:
+    # "language" is the 3-letter code used in RapidPro
+    # "code" is the 2 letter code used in CrowdIn
+    languages: list[dict]
+    # Git repository (synched with crowdin) to read translation PO files from
+    translation_repo: str
+    # Folder within within the `translation_repo` repository to read translation PO files from
+    folder_within_repo: str
+    # Not Implemented: Commit hash or tag in the repo
+    # TODO: Offer branch, and then store the commit hash as part of the meta info about the output
+    commit_hash: str = None
+    commit_tag: str = None
+
+
+SOURCE_CONFIGS = {   
+    "sheets" : SheetsSourceConfig,
+    "json" : JSONSourceConfig,
+    "translation_repo" : TranslationSourceConfig,
+    "safeguarding" : SafeguardingSourceConfig,
+}
+
+
+@dataclass(kw_only=True)
+class Config:
+    meta: dict
+    parents: list[dict] = None
```

Comment:
I was thinking that sources would be a single list containing all of the spreadsheets required to create the flows. It could include spreadsheets from other repositories. The order in which the spreadsheets appear in the list would dictate how sheets within them might be merged or replaced. If we then push as much information as possible into the spreadsheets, it would all be subject to the same rules regarding inheritance.

I used to think that it would be useful to be able to explicitly configure which steps are included in the pipeline, but over time I've come to realise that the steps are pretty much the same for every deployment, and that a step can decide whether it should run or not based on whether it has sufficient information (in config or spreadsheets) to do its thing.